### ISSUE TITLE
Make reset pin optional

### ISSUE THREAD

**Issue by thijstriemstra** [wollewald/MCP23017_WE#4]
I just found out the MCP203017 reset pin was hooked up to a different pin on the ESP32 than one I specified. But it doesn't seem to have made a difference, everything properly worked anyway. So is the reset pin optional?

**Comment by wollewald** [comment]
The reset function is only used by the Init function to ensure that the MCP23017 starts with a defined status. Otherwise, if you reset the ESP32, you might not reset the MCP23017. And then it could still "remember" its status before the ESP32 was reset and cause problems. And of course you need the reset pin if you want to use the reset function.

**Comment by thijstriemstra** [comment]
But powering down the ESP32 and MCP23017 will reset the MCP23017 right? Unlike the ESP32 reset button?

**Comment by wollewald** [comment]
Yes that's right.

**Comment by thijstriemstra** [comment]
thanks!

**Comment by thijstriemstra** [comment]
oh closed, too soon. For now I modified the library to not change the `pinMode` for the reset pin (e.g. when i pass in -1) but it would be nice to make the reset pin officially optional. So maybe the default should be -1 which indicates it's optional. I wouldn't change the examples, only add a note there saying it's optional.

I commented this out for now:

```cpp
void MCP23017::Init(){
	//pinMode(resetPin, OUTPUT); 
	//reset();
	ioConA = B00000000;
        // etc..
```
